modules/schema/src/main/java/org/apache/ignite/internal/schema/DecimalNativeType.java

```diff
@@ -60,8 +60,7 @@ public int scale() {
     /** {@inheritDoc} */
     @Override public boolean mismatch(NativeType type) {
         return super.mismatch(type)
-            || precision < ((NumericNativeType)type).precision
-            || scale < ((NumericNativeType)type).scale;
+            || precision < ((DecimalNativeType)type).precision;
```

Comment:
Agree, we are ok to round up numbers that has higher scale.
But what if user pass a number of higher precision due to it high scale?

E.g. table column type is Decimal(8,2), but user pass 123.123456789.
We detect user number as of Decimal(12,9), but actually, it perfectly fits to the column type after round up.

Have I missed smth?

Reply:
No, I missed this case. Thanks, will think, how to fix it.

Reply:
I decided to check precision after scaling in RowAssembler.